### PR TITLE
New MCQ feature added

### PR DESCRIPTION
**Pull request by Yashig23**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/quml-library/src/lib/sa/sa.component.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ export class SaComponent implements OnInit, OnChanges, AfterViewInit {
   constructor( public domSanitizer: DomSanitizer, private utilService: UtilService ) { }
 
   ngOnInit() {
+    console.log(this.questions);
```
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/sa/sa.component.ts]: remove it

File: projects/quml-library/src/lib/icon/hint/hint.component.css
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/icon/hint/hint.component.css:1]: remove it

File: projects/quml-library/src/lib/header/header.component.ts
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/header/header.component.ts:1]: undo it

File: projects/quml-library/src/lib/section-player/section-player.component.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { eventName, pageId, TelemetryType, Cardinality, QuestionType } from '../
 import { DEFAULT_SCORE } from '../player-constants';
 import { UtilService } from '../util-service';
 
+
```
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/section-player/section-player.component.ts]: remove it

File: projects/quml-library/src/lib/section-player/section-player.component.ts [hunk 3]
```diff
@@ -29,6 +30,8 @@ export class SectionPlayerComponent implements OnChanges, AfterViewInit {
   @Output() showScoreBoard = new EventEmitter<any>();
 
   @ViewChild('myCarousel', { static: false }) myCarousel: CarouselComponent;
+ // @ViewChild('myCarousel', { static: false }) myCarousel: SlickCarouselComponent;
```
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/section-player/section-player.component.ts]: remove it

File: projects/quml-library/src/lib/section-player/section-player.component.ts [hunk 4]
```diff
@@ -880,10 +915,10 @@ export class SectionPlayerComponent implements OnChanges, AfterViewInit {
 
   onAnswerKeyDown(event: KeyboardEvent) {
     /* istanbul ignore else */
-    if (event.key === 'Enter') {
+    /*if (event.key === 'Enter') {
```
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/section-player/section-player.component.ts]: undo it

Comment (Yashig23) [reply projects/quml-library/src/lib/section-player/section-player.component.ts]: completed it

File: projects/quml-library/src/lib/section-player/section-player.component.ts (hunk 3 above)
Comment (Yashig23) [reply projects/quml-library/src/lib/section-player/section-player.component.ts]: removed

File: projects/quml-library/src/lib/section-player/section-player.component.ts (hunk 2 above)
Comment (Yashig23) [reply projects/quml-library/src/lib/section-player/section-player.component.ts]: removed

File: projects/quml-library/src/lib/header/header.component.ts
Comment (Yashig23) [reply projects/quml-library/src/lib/header/header.component.ts:1]: completed it

File: projects/quml-library/src/lib/sa/sa.component.ts (hunk 1 above)
Comment (Yashig23) [reply projects/quml-library/src/lib/sa/sa.component.ts]: removed

File: projects/quml-library/src/lib/assets/styles/quml-carousel.css
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/assets/styles/quml-carousel.css:1]: remove empty spaces

File: projects/quml-library/src/lib/attachments/attachments.component.ts
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/attachments/attachments.component.ts:1]: remove unnecessary commnets for variables

File: projects/quml-library/src/lib/mcq/mcq.component.html
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/mcq/mcq.component.html:1]: this file changes are completely unnecessary, remove changes.

File: projects/quml-library/src/lib/sa/sa.component.scss
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/sa/sa.component.scss:1]: unnecessary change

File: projects/quml-library/src/lib/services/viewer-service/viewer-service.ts
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/services/viewer-service/viewer-service.ts:1]: unneccessary change

File: projects/quml-library/src/lib/slider-question/slider-question.component.ts
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/slider-question/slider-question.component.ts:1]: remove the unnecessary comments for variables, keep consistency in commenting

File: projects/quml-player-wc/src/app/app.module.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { createCustomElement } from '@angular/elements';
 import { BrowserModule } from '@angular/platform-browser';
 import { SunbirdPlayerSdkModule } from '@project-sunbird/sunbird-player-sdk-v9';
 import { CarouselModule } from 'ngx-bootstrap/carousel';
+
```
Comment (ramkumar-ns-dev) [inline projects/quml-player-wc/src/app/app.module.ts]: unnecessary change

File: projects/quml-library/src/lib/assets/styles/quml-carousel.css
Comment (Yashig23) [reply projects/quml-library/src/lib/assets/styles/quml-carousel.css:1]: completed

File: projects/quml-library/src/lib/attachments/attachments.component.ts
Comment (Yashig23) [reply projects/quml-library/src/lib/attachments/attachments.component.ts:1]: completed

File: projects/quml-library/src/lib/mcq/mcq.component.html
Comment (Yashig23) [reply projects/quml-library/src/lib/mcq/mcq.component.html:1]: completed

File: projects/quml-player-wc/src/app/app.module.ts (hunk 5 above)
Comment (Yashig23) [reply projects/quml-player-wc/src/app/app.module.ts]: completed

File: projects/quml-library/src/lib/services/viewer-service/viewer-service.ts
Comment (Yashig23) [reply projects/quml-library/src/lib/services/viewer-service/viewer-service.ts:1]: completed

File: projects/quml-library/src/lib/slider-question/slider-question.component.ts
Comment (Yashig23) [reply projects/quml-library/src/lib/slider-question/slider-question.component.ts:1]: completed

**Review by ramkumar-ns-dev — APPROVED** [review]

File: projects/quml-library/src/lib/new-mcq-question/new-mcq-question.component.ts [hunk 6]
```diff
@@ -1,43 +1,70 @@
...
+options: any; 
+value: any; 
+label: any; 
+hints: any; 
+sizeLimit: any; 
 
-  onclick(_value: string){
-    console.log(_value);
+// This method is called when the component is initialized
+ngOnInit() {
```
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/new-mcq-question/new-mcq-question.component.ts:42]: remove comments for life cycles

File: projects/quml-library/src/lib/new-mcq-question/new-mcq-question.component.ts [hunk 7]
```diff
@@ -1,43 +1,70 @@
...
+    this.options = this.questions?.interactions.response1.options;
+    this.value = this.questions?.interactions.response1.options.value;
+    this.hints = this.questions?.hint;
+    this.label = this.questions?.interactions.response1.options.label;
+    this.sizeLimit = this.questions?.evidence.sizeLimit;
   }
 }
+
+// Event handler for a click event
+onclick(_value: string) {
```
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/new-mcq-question/new-mcq-question.component.ts]: name the function properly and give proper comments

File: projects/quml-library/src/lib/section-player/section-player.component.html [hunk 8]
```diff
@@ -30,13 +33,15 @@
               [contentName]="showRootInstruction ? parentConfig?.contentName : parentConfig?.isSectionsAvailable ? sectionConfig?.metadata?.name : parentConfig?.contentName">
             </quml-startpage>
           </slide>
+          <!--</ngx-carousel-item>-->
```
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/section-player/section-player.component.html]: remove the commented codes in all the places

File: projects/quml-library/src/lib/services/viewer-service/viewer-service.ts
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/services/viewer-service/viewer-service.ts:1]: as i'm saying there is no change in this file

File: projects/quml-library/src/lib/slider-question/slider-question.component.ts [hunk 9]
```diff
@@ -40,14 +51,37 @@ export class SliderQuestionComponent implements OnInit {
 
   constructor( public domSanitizer: DomSanitizer, private utilService: UtilService ) { }
 
+  
+  // Initializes the component with question data.
```
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/slider-question/slider-question.component.ts]: remove life cycles function commands

File: projects/quml-library/src/lib/terms-condition-box/terms-condition-box.component.html [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+              <input type="checkbox" class="form-check-input" name="termsField" ngModel />
+              <p class="termsContent">I confirm that the Content submitted by me comply with the terms and policies of Sunbird, including <a href="#">Sunbird's Content Policy</a>. I have made sure that I do not violate others' copyright, privacy, or third party rights.</p>
+          </label>
+          <!-- Buttons for the dialog box -->
+          <section class="buttons-for-termBox">
+              <!-- Button to cancel and not upload -->
+              <button type="button" (click)="cancelTermsDialog()" class="doNotUplaodBtn">Do not Upload</button>
+              <!-- Button to submit and upload with conditions -->
+              <button type="submit" class="uploadBtn" [disabled]="termsForm.value.termsField!=true" (click)="inputFileField.click()">Upload</button>
+              <!-- <button type="submit" class="uploadBtn" [disabled]="termsForm.value.termsField!=true" (click)="triggerInputClick()">Upload</button> -->
```
Comment (ramkumar-ns-dev) [inline projects/quml-library/src/lib/terms-condition-box/terms-condition-box.component.html]: remove commented garbage code in all over the PR.

Comment (Yashig23) [reply projects/quml-library/src/lib/terms-condition-box/terms-condition-box.component.html]: completed

File: projects/quml-library/src/lib/section-player/section-player.component.html (hunk 8 above)
Comment (Yashig23) [reply projects/quml-library/src/lib/section-player/section-player.component.html]: completed

File: projects/quml-library/src/lib/new-mcq-question/new-mcq-question.component.ts (hunk 7 above)
Comment (Yashig23) [reply projects/quml-library/src/lib/new-mcq-question/new-mcq-question.component.ts]: completed

File: projects/quml-library/src/lib/new-mcq-question/new-mcq-question.component.ts (hunk 6 above)
Comment (Yashig23) [reply projects/quml-library/src/lib/new-mcq-question/new-mcq-question.component.ts:42]: completed
